src/guide/advanced/isi.md

```diff
@@ -140,47 +99,23 @@ a group of permissions (a "role"). Granted roles and permissions can only
 be removed via the `Revoke` instruction. As such, these instructions should
 be used carefully.
 
-### `SetKeyValue`/`RemoveKeyValue`
+## `SetKeyValue`/`RemoveKeyValue`
 
 These instructions are used with the key/value
 [`Store` asset type](../objects/metadata.md#store-asset). This use case has
 not received much attention so far, because storing data in the blockchain
 is a rather advanced topic that we shall cover separately.
 
-### Query
-
-We talk extensively about queries in the [dedicated section](queries.md)
-where all the queries that can be made from the client-side are listed.
-This is not necessarily the only kind of information that is available on
-the network, but it's the only kind of information that is _guaranteed_ to
-be accessible on all networks.
-
-For each deployment of Iroha, there might be other available information.
-For example, the availability of telemetry data is up to the network
-administrators. It's entirely their decision whether or not they want to
-allocate processing power to track the work instead of using it to do the
-actual work. By contrast, some functions are always required, e.g. having
-access to your account balance.
-
-### Expressions, Conditionals, Logic
-
-All ISI operate on expressions. Each expression has an `EvaluatesTo`, which
-is used in instruction execution. While you could specify the account name
-directly, you could also specify the account ID via some mathematical or
-string operation. You can check if an account is registered on the
-blockchain too.
-
-Using expressions that implement `EvaluatesTo<bool>`, you can set up
-conditional logic and execute more sophisticated operations on-chain. For
-example, you can submit a `Mint` instruction only if a specific account is
-registered.
-
-Recall that you can combine this with queries, and as such can program the
-blockchain to do some amazing stuff. This is what we refer to as _smart
-contracts_, the defining feature of the advanced usage of blockchain
-technology.
-
-[^1]:
-    For prospective wizards, the
-    [Iroha 2 Whitepaper](https://github.com/hyperledger/iroha/blob/iroha2-dev/docs/source/iroha_2_whitepaper.md)
-    is a good start.
+## `ExecuteTrigger`
+
+This instruction is used to execute [triggers](./triggers.md).
+
+## Composite instructions
+
+Iroha also offers composite instructions (`If`, `Pair`, `Sequence`) to
+execute instructions in a certain way:
```

Comment:
```suggestion
be executed in a certain way:
```

Reply:
composite instructions execute other instructions in a specified way, so I don't think passive voice is applicable here